src/main/scala/beam/agentsim/agents/vehicles/BeamVehicle.scala

```diff
@@ -47,160 +58,153 @@ trait VehicleData extends BeamAgentData with Vehicle {
 }
 
 object BeamVehicle {
+  val ActorPrefixName = "vehicle-"
+
+  def props(vehicleId: Id[Vehicle], matSimVehicle: Vehicle, powertrain: Powertrain) = {
+    Props(classOf[BeamVehicle], vehicleId, VehicleData.vehicle2vehicleData(matSimVehicle), powertrain, None)
```

Comment:
You're probably right on this. Either we only define props in concretes or we pass the class in here.